sandbox/src/components/MessageFeedDemo/MessageFeed.js

```diff
@@ -0,0 +1,628 @@
+          eventIndex: 0
+        },
+        {
+          payload: [
+            {
+              scope: "Target",
+              hint: "35",
+              ttlSeconds: 1800
+            },
+            {
+              scope: "AAM",
+              hint: "9",
+              ttlSeconds: 1800
+            },
+            {
+              scope: "EdgeNetwork",
+              hint: "or2",
+              ttlSeconds: 1800
+            }
+          ],
+          type: "locationHint:result"
+        },
+        {
+          payload: [
+            {
+              key: "kndctr_4DA0571C5FDC4BF70A495FC2_AdobeOrg_cluster",
+              value: "or2",
+              maxAge: 1800,
+              attrs: {
+                SameSite: "None"
+              }
+            }
+          ],
+          type: "state:store"
+        }
+      ]
+    };
+    async function applyRuleEngineResponse() {
+      window.alloy("applyResponse", {
+        renderDecisions: true,
+        decisionContext: {
+          color: "pink",
+          action: "lipstick"
+        },
+        responseBody: mockResponse
+      });
+    }
+    document
+      .getElementById("social-media-share")
+      .addEventListener("click", evt => {
+        evt.stopImmediatePropagation();
+        evt.preventDefault();
+
+        window.alloy("evaluateRulesets", {
+          action: "share-social-media"
+        });
+      });
+
+    document.getElementById("deposit-funds").addEventListener("click", evt => {
+      evt.stopImmediatePropagation();
+      evt.preventDefault();
+
+      window.alloy("evaluateRulesets", {
+        action: "deposit-funds"
+      });
+    });
+
+    document.getElementById("magic").addEventListener("click", evt => {
+      evt.stopImmediatePropagation();
+      evt.preventDefault();
+
+      window.alloy("evaluateRulesets", {
+        "~type": "com.adobe.eventType.generic.track",
+        "~source": "com.adobe.eventSource.requestContent",
+        action: "hello"
+      });
+    });
+
+    let isMounted = true;
+    const fetchData = async () => {
+      try {
+        const result = await window.alloy("subscribeMessageFeed", {
+          surface: "web://target.jasonwaters.dev/aep.html",
+          callback: ({ items = [], clicked, rendered }) => {
+            if (isMounted) {
+              setMessageFeedItems(items);
+            }
+            clicked(items);
+            rendered(items);
+          }
+        });
+        if (isMounted) {
+          await applyRuleEngineResponse(result);
+        }
+      } catch (error) {
+        console.error("Error:", error);
+      }
+    };
+    fetchData();
+    return () => {
+      isMounted = false;
+    };
+  }, []);
+
+  return (
+    <div>
+      <ContentSecurityPolicy />
+      <div style={{ margin: "10px 0" }}>
+        <button id="social-media-share">Share on social media</button>
+        <button id="deposit-funds">Deposit funds</button>
+        <button id="magic">Abracadabra 🪄</button>
```

Comment:
I suggest removing the Abracadabra button since it opens an IAM -- it's irrelevant to testing message feed

Reply:
It might be nice to repurpose the Abracadabra button to instead be a "Reset" button.  Pressing it would clear local storage and reload the page.  This would be useful for demos and tests so it's easier to clear state and validate message feed items.

Reply:
done.